[iOS26][Shell] Updated nav bar appearance logic for iOS26

kubaflo opened the pull request:
<!-- Please let the below note in for people that find this PR -->
> [!NOTE]
> Are you waiting for the changes in this PR to be merged?
> It would be very helpful if you could [test the resulting artifacts](https://github.com/dotnet/maui/wiki/Testing-PR-Builds) from this PR and let us know in a comment if this change resolves your issue. Thank you!

<!--
!!!!!!! MAIN IS THE ONLY ACTIVE BRANCH. MAKE SURE THIS PR IS TARGETING MAIN. !!!!!!! 
-->

### Description of Change

Refines the condition for configuring the navigation bar appearance by checking the iOS version and background color. This ensures correct appearance behavior on different iOS versions.


|Before|After|
|--|--|
|<video src="https://github.com/user-attachments/assets/a525b9cc-68ff-4b5f-9eb5-a98e9b717763" width="300px"></video>|<video src="https://github.com/user-attachments/assets/402c2ab7-a46a-4452-90f0-19c62ae5cea2" width="300px"></video>|

<!--
Are you targeting main? All PRs should target the main branch unless otherwise noted.
-->

kubaflo committed:
Updated nav bar appearance logic for iOS26

Refines the condition for configuring the navigation bar appearance by checking the iOS version and background color. This ensures correct appearance behavior on different iOS versions.

jsuarezruiz reviewed (changes requested):
Could include a related test?

kubaflo commented on src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellNavBarAppearanceTracker.cs:
```diff
@@ -101,7 +101,7 @@ void UpdateiOS13NavigationBarAppearance(UINavigationController controller, Shell
 				navBar.Translucent = true;
 			}
-			else
+			else if (!OperatingSystem.IsIOSVersionAtLeast(26) || appearance.BackgroundColor is not null)
```
Great idea, thanks!

kubaflo committed:
Add comment for iOS 26 nav bar appearance logic

Added explanatory comments regarding the handling of navigation bar appearance on iOS 26 and later, clarifying why opaque backgrounds are only forced when a background color is set.
